securedrop/journalist_app/api.py

```diff
@@ -224,43 +221,15 @@ def all_source_replies(source_uuid: str) -> Tuple[flask.Response, int]:
             if not data["reply"]:
                 abort(400, "reply should not be empty")
 
-            source.interaction_count += 1
             try:
-                filename = Storage.get_default().save_pre_encrypted_reply(
-                    source.filesystem_id,
-                    source.interaction_count,
-                    source.journalist_filename,
-                    data["reply"],
-                )
-            except NotEncrypted:
-                return jsonify({"message": "You must encrypt replies client side"}), 400
-
-            # issue #3918
-            filename = path.basename(filename)
-
-            reply = Reply(session.get_user(), source, filename, Storage.get_default())
-
-            reply_uuid = data.get("uuid", None)
-            if reply_uuid is not None:
-                # check that is is parseable
-                try:
-                    UUID(reply_uuid)
-                except ValueError:
-                    abort(400, "'uuid' was not a valid UUID")
-                reply.uuid = reply_uuid
-
-            try:
-                db.session.add(reply)
-                seen_reply = SeenReply(reply=reply, journalist=session.get_user())
-                db.session.add(seen_reply)
-                db.session.add(source)
-                db.session.commit()
+                reply = utils.save_reply(source, data)
```

Comment:
I'd like for the v2 API event-handlers to follow this pattern, as much as possible reusing "write to database" functions factored out from the v1 API endpoint.  Any concerns about this approach?